spec.md

```diff
@@ -202,11 +202,11 @@ help intermediate gateways determine how to route the events.
 
 ### source
 * Type: `URI-reference`
-* Description: This describes the event producer. Often this will include
-  information such as the type of the event source, the organization
-  publishing the event, the process that produced the event, and some unique
-  identifiers. The exact syntax and semantics behind the data encoded in the URI
-  is event producer defined.
```

Comment:
Why are you dropping these two sentences? They give a lot of motivation for why a URI is chosen at all. If you don't want to `include information such as...`, then you could also go straight for a UUID and be done with it 😉 

I think it is very valuable to describe what to put into the `source`, and why a hierarchical data structure was chosen.

Reply:
To me, the important thing here is that source can be a URI with an internet-unique authority so you don't have to resort to UUIDs for uniqueness. As I read it, the only normative use of the source is to uniquely identify a producer.

The design of source URIs will depend on the application. It might include "information such as type/organization/process..." but it might not. This spec doesn't seem a good place for advice on URI design, which is a topic in its own right.

I'm ok with putting it back if you feel strongly about it.

Reply:
I prefer the existing text because it's a little less strict in that the value doesn't necessarily have to be a valid address on the web. For example, they could use `http://myserver/....` is that is ok for their setup. We purposely didn't want to get too precise here to allow for flexibility.

Reply:
Check my latest text. I want to make it clear that you MAY have an internet-unique authority and URI here but you also MAY also have something of application specific scope. Otherwise it's hard to answer the original question "what is the scope in which source+id is unique" in any meaningful way.

Reply:
@alanconway can you elaborate on what problem you're trying to solve with this change? In the end, aren't these all just opaque identifiers since we're not asking the receiver to do anything with these URI-references - like re-reference them?

Reply:
I'm trying to clarify the scope of uniqueness, the "U" in URI. URIs have a standard authority component which provides well-known, internet-wide uniqueness guarantees. I'm trying to clarify that a "source" can be URI-unique if it has a URI authority OR it can be an authority-less reference path which is only unique in some application-defined context. Both are useful uniqueness guarantees for different applications. 

We could leave it unstated since  it's implied by using a URI-reference, but this PR is requesting "Clarify scope of eventID's uniqueness" so it seems to need saying.

Reply:
My interpretation of the request for clarity around uniqueness was more around simple string comparison type of things :-)  So, while the semantics behind these values might be interesting in some cases, I think what's more important (from a spec perspective) is that people know if they do the equivalent of `ce.source + ce.id` they'll get a unique value within the scope of that producer, and can use it appropriately.  Getting into the semantics of the strings, while interesting, doesn't really change what the code a consumer would write - I don't think anyway.

Reply:
+1 I've restored the original comments on "source" - with the exception of saying "identifies" rather than "describes" and changed the "id" part along the lines you suggested.